src/main/java/com/limechain/trie/structure/node/handle/NodeHandle.java

```diff
@@ -25,4 +26,8 @@ public List<Nibble> getPartialKey() {
         //TODO: implement
         throw new NotImplementedException("Not yet implemented.");
     }
+
+    public T getUserData() {
+        return trieStructure.nodeAtIndex(new TrieNodeIndex(nodeIndex)).getUserData();
```

Comment:
That seems to be an infinite recursion... we'll have to refactor the access modifiers a bit, so that a `NodeHandle` has direct access to the `TrieStructure`'s underlying data structure, containing the actual structural nodes.

Reply:
For now we can do something like
```java
    return trieStructure.getUserDataAtIndex(new TrieNodeIndex(nodeIndex));
```
and perhaps we could also store the actual integer wrapped in a `NodeIndex` within the `NodeHandle`... or maybe abolish the notion of a `NodeIndex` if it becomes too messy.